Zend/zend_execute_API.c

```diff
@@ -438,6 +438,8 @@ void shutdown_executor(void) /* {{{ */
 	zval *zv;
 #if ZEND_DEBUG
 	bool fast_shutdown = 0;
+#elif defined(__SANITIZE_ADDRESS__)
+	bool fast_shutdown = !EG(full_tables_cleanup);
 #else
 	bool fast_shutdown = is_zend_mm() && !EG(full_tables_cleanup);
```

Comment:
fast_shutdown relays on Zend MM that frees the whole request heap at once. It doesn't make sense without it and will cause memory leaks.